### ISSUE TITLE
Modal accessibility

### ISSUE THREAD

**Issue by rockycodes** [civiform/civiform#4196]
**Describe the bug**
When a modal opens on the page, the focus does not shift to the modal and the user has to tab through the elements behind the modal before being able to interact with the modal. This could make for a confusing experience for folks using screen readers.

Also, the modal fails the accessibility test [here](https://github.com/civiform/civiform/blob/7f29b40800422ec8b854c7d2ac9ab783f9afa420/browser-test/src/support/index.ts#L476) with the message that not all content is contained within landmarks (https://dequeuniversity.com/rules/axe/4.6/region?application=axeAPI)

**To Reproduce**

1. Go to the block edit view
2. Use the tab button to navigate the page
3. Hit "Enter" when you have the "Delete screen" button selected
4. Use the tab button to continue navigating the page
5. You will have to tab through all elements behind the modal before getting to the modal

**Expected behavior**
When a user opens the modal and continues navigating the page via manual tabbing the elements on the modal should be the next ones selected.

**Screenshots**
You can see that "Edit visibility condition" is selected when I hit "tab" after opening the modal
![image](https://user-images.githubusercontent.com/24319951/217978387-6d02257e-1b42-405f-851d-391b41fe3e65.png)

**Desktop (please complete the following information):**

- OS: iOS
- Browser Firefox
- Version 109.0.1

**Additional context**

**Comment by nb1701** [comment]
Need to investigate if there are applicant-face modals.  If it's only admin, we probably don't need to do it, but if there are applicant-facing ones, we should do this soon.

**Comment by dkatzz** [comment]
> Need to investigate if there are applicant-facing modals. If it's only admin, we probably don't need to do it, but if there are applicant-facing ones, we should do this soon.
> 
> Edit: May be some with the eligibility and address correction features. @gwendolyngoetz @seanmalbert @dkatzz can you take a look and see if this is an issue with modals for those features?

There aren't any modals for general eligibility, but I believe there are modals for the address correction feature. @seanmalbert and @gwendolyngoetz to confirm.

**Comment by caitlinshk** [comment]
This is also an issue with the "Create account or sign in" modal, which is applicant-facing.

![Image](https://github.com/civiform/civiform/assets/147177873/f9c6d798-fe3e-4db3-b543-608e190149cd)
